app/src/main/java/com/example/myapplication/mvp/presenter/CalculatorPresenter.kt

```diff
@@ -5,55 +5,110 @@ import com.example.myapplication.mvp.view.CalculatorView
 import com.example.myapplication.utils.DIVIDE
 import com.example.myapplication.utils.EMPTY_STRING
 import com.example.myapplication.utils.MULTIPLY
+import com.example.myapplication.utils.ONE_INT
+import com.example.myapplication.utils.ZERO_FLOAT_RESULT
 import com.example.myapplication.utils.PLUS
+import com.example.myapplication.utils.POINT
 import com.example.myapplication.utils.SUBTRACT
+import com.example.myapplication.utils.TOAST_ZERO_DIVIDE
+import com.example.myapplication.utils.ZERO_INT
 
-class CalculatorPresenter(val model: CalculatorModel, val view: CalculatorView) {
+open class CalculatorPresenter(val model: CalculatorModel, val view: CalculatorView) {
 
     fun onNumberPressed(number: String) {
-        if (model.getOperand() == EMPTY_STRING) {
-            model.setOperatorOne("${model.getOperatorOne()}$number")
-            view.setVisor(model.getOperatorOne())
+        if (model.result == ZERO_FLOAT_RESULT) {
+            if (model.operatorOne.isEmpty()) {
+                model.operatorOne = "${model.operatorOne}$number"
+                view.setVisor(model.operatorOne)
+            } else {
+                model.operatorTwo = "${model.operatorTwo}$number"
+                view.setVisor("${model.operatorOne}${model.operand}${model.operatorTwo}")
+            }
+            if (number == POINT)
+                view.diseablePoint()
         } else {
-            model.setOperatorTwo("${model.getOperatorTwo()}$number")
-            view.setVisor("${model.getOperatorOne()}${model.getOperand()}${model.getOperatorTwo()}")
+            model.result = ZERO_FLOAT_RESULT
+            model.operatorOne = "${model.operatorOne}$number"
+            view.setVisor(model.operatorOne)
         }
     }
 
     fun onOperatorPressed(operator: String) {
-        if (model.getOperatorOne() != EMPTY_STRING) {
-            if (model.getOperand() == EMPTY_STRING) {
-                model.setOperand(operator)
-                view.setVisor("${model.getOperatorOne()}$operator")
-            } else {
-                model.setOperatorTwo(EMPTY_STRING)
-                model.setOperand(operator)
-                view.setVisor("${model.getOperatorOne()}$operator")
+        if (model.result == ZERO_FLOAT_RESULT) {
+            if (model.operatorOne.isNotEmpty() && model.operatorTwo.isEmpty()) {
+                if (model.operand.isEmpty()) {
+                    model.operand = operator
+                    view.setVisor("${model.operatorOne}$operator")
```

Comment:
this code is duplicated whether is empty or not. You may want to change the logic to:
```
if (model.operand.isNotEmpty()) {
  model.operatorTwo = EMPTY_STRING
}
model.operand = operator
view.setVisor("${model.operatorOne}$operator")
```

Reply:
done